### DISCUSSION TITLE
Quit corfu completion when leaving insert state

### DISCUSSION THREAD

**Discussion by witek** [meow-edit/meow#422] [Q&A] [answered]
I am using `corfu` for completions. It shows the popup in insert state and it quits when I select a candidate. But it does not quit when I `<escape>` back to normal state. It still shows the popup with candidates.

There is the command `corfu-quit` which can quit the popup. But how to integrate it into `meow`?

**Comment by schrenker** [comment]
I have defined this for myself and works well:
`  (defun schrenker/meow-true-escape ()
    (interactive)
    (call-interactively #'meow-insert-exit)
    (call-interactively #'corfu-quit))
`

Then I bind this to esc:
  (define-key meow-insert-state-keymap [escape] 'schrenker/meow-true-escape)

**Comment by DogLooksGood** [comment, marked as answer]
There's a hook `meow-insert-exit-hook`.

```
(add-hook 'meow-insert-exit-hook 'corfu-quit)
```
